Classes/Netlogix/Cqrs/Log/CommandLogger.php

```diff
@@ -23,9 +30,12 @@ class CommandLogger {
 	 * Log the given command
 	 *
 	 * @param AbstractCommand $command
+	 * @param \Exception $exception
 	 */
-	public function logCommand(AbstractCommand $command) {
+	public function logCommand(AbstractCommand $command, \Exception $exception = null) {
 		$commandLogEntry = new CommandLogEntry($command);
+		$commandLogEntry->setException($exception);
 		$this->commandLogEntryRepository->add($commandLogEntry);
+		$this->persistenceManager->persistAll();
```

Comment:
In case of an exception persistAll is not called for a good reason: something broke, don't persist an inconsistent state. Therefore we need some other solution to persist this log entry. Maybe persistenceManager->whitelistObject() works, otherwise we might need to go down to doctrine for this.